pycbc/noise/reproduceable.py

```diff
@@ -171,7 +171,9 @@ def colored_noise(psd, start_time, end_time,
                          end_time + filter_duration,
                          seed=seed,
                          sample_rate=sample_rate)
+    # make sure FD white_noise has the same length as asd
     white_noise = white_noise.to_frequencyseries()
+    white_noise.resize(len(asd))
```

Comment:
I'm not sure this is exactly what we want to do here. Maybe the fix should be in the function just above. We definitely can't just resize the freuqency-domain object, but that distorts the sample rate (perhaps only slightly if the vector is long, but still not quite right).

Reply:
I don't understand the problem this is trying to solve. This function takes `start_time`, `end_time` and `sample_rate`, which completely specifies the noise (and so we shouldn't do this).

But if this is a problem, it implies an issue above, as we do do a `psd.resize` previously to make things match.

Can you give an example of a failure?